mixer/tools/mixgen/cmd/api_test.go

```diff
@@ -62,41 +60,49 @@ func TestGenerator_Generate(t *testing.T) {
 			"testdata/apa/template_instance.proto.golden"},
 	}
 	for _, v := range tests {
-		t.Run(v.name, func(t *testing.T) {
+		t.Run(v.name, func(tt *testing.T) {
 			tmpDir := path.Join(os.TempDir(), v.name)
 			_ = os.MkdirAll(tmpDir, os.ModeDir|os.ModePerm)
 			oIntface, err := os.Create(path.Join(tmpDir, path.Base(v.wantIntFace)))
 			if err != nil {
-				t.Fatal(err)
+				tt.Fatal(err)
 			}
 			oTmpl, err := os.Create(path.Join(tmpDir, path.Base(v.wantProto)))
 			if err != nil {
-				t.Fatal(err)
+				tt.Fatal(err)
 			}
 
 			defer func() {
-				if !t.Failed() {
+				if !tt.Failed() {
 					if removeErr := os.Remove(oIntface.Name()); removeErr != nil {
-						t.Logf("Could not remove temporary file %s: %v", oIntface.Name(), removeErr)
+						tt.Logf("Could not remove temporary file %s: %v", oIntface.Name(), removeErr)
 					}
 					if removeErr := os.Remove(oTmpl.Name()); removeErr != nil {
-						t.Logf("Could not remove temporary file %s: %v", oTmpl.Name(), removeErr)
+						tt.Logf("Could not remove temporary file %s: %v", oTmpl.Name(), removeErr)
 					}
 				}
 			}()
 
-			g := Generator{OutInterfacePath: oIntface.Name(), OAugmentedTmplPath: oTmpl.Name(), ImptMap: importmap}
-
-			if err := g.Generate(v.descriptor); err != nil {
-				t.Fatalf("Generate(%s) produced an error: %v", v.descriptor, err)
+			args := []string{"api", "-t", v.descriptor, "--out_go", oIntface.Name(), "--out_proto", oTmpl.Name()}
+			for _, m := range importmap {
```

Comment:
question: does this ever change, or should we just make one big default string (instead of running through this loop) ?